src/python/pants/backend/python/dependency_inference/module_mapper_test.py

```diff
@@ -226,74 +210,67 @@ def test_map_first_party_modules_to_addresses(rule_runner: RuleRunner) -> None:
+                ModuleProvider(
+                    Address("src/python/project/util", relative_file_path="tarutil.py"),
+                    ModuleProviderType.IMPL,
+                ),
             ),
             "project_test.demo_test": (
-                Address("tests/python/project_test/demo_test", relative_file_path="__init__.py"),
+                ModuleProvider(
+                    Address(
+                        "tests/python/project_test/demo_test", relative_file_path="__init__.py"
+                    ),
+                    ModuleProviderType.IMPL,
+                ),
             ),
             "protos.f1_pb2": (
-                Address("src/python/protos", relative_file_path="f1.proto", target_name="protos"),
-            ),
-            "stubs.stub": (
-                Address("src/python/stubs", relative_file_path="stub.py"),
-                Address("src/python/stubs", relative_file_path="stub.pyi"),
+                ModuleProvider(
+                    Address(
+                        "src/python/protos", relative_file_path="f1.proto", target_name="protos"
+                    ),
+                    ModuleProviderType.IMPL,
+                ),
             ),
-        }
-    )
-    assert result.ambiguous_modules == FrozenDict(
-        {
             "protos.f2_pb2": (
-                Address("src/python/protos", target_name="py"),
-                Address("src/python/protos", relative_file_path="f2.proto", target_name="protos"),
-            ),
-            "protos_ambiguous.f_pb2": (
-                Address("src/python/protos_ambiguous", target_name="py"),
-                Address(
-                    "src/python/protos_ambiguous",
-                    relative_file_path="f.proto",
-                    target_name="protos1",
+                ModuleProvider(
+                    Address("src/python/protos", target_name="py"), ModuleProviderType.IMPL
                 ),
-                Address(
-                    "src/python/protos_ambiguous",
-                    relative_file_path="f.proto",
-                    target_name="protos2",
+                ModuleProvider(
+                    Address(
+                        "src/python/protos", relative_file_path="f2.proto", target_name="protos"
+                    ),
+                    ModuleProviderType.IMPL,
                 ),
             ),
-            "stub_ambiguity.f": (
-                Address("build-support/stub_ambiguity", relative_file_path="f.pyi"),
-                Address("src/python/stub_ambiguity", relative_file_path="f.pyi"),
-            ),
-            "two_owners": (
-                Address("build-support", relative_file_path="two_owners.py"),
-                Address("src/python", relative_file_path="two_owners.py"),
-            ),
         }
     )
-    assert result.modules_with_type_stub == FrozenOrderedSet(["stubs.stub"])
 
 
 def test_map_third_party_modules_to_addresses(rule_runner: RuleRunner) -> None:
```

Comment:
This function was simplified - there's no concept of "ambiguity" with this codepath anymore, so fewer edge cases to test.